subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1162,10 +1162,12 @@ private DefaultConfiguration newConfiguration() {
         String newName = getNameWithCopySuffix();
 
         Factory<ResolutionStrategyInternal> childResolutionStrategy = resolutionStrategy != null ? Factories.constant(resolutionStrategy.copy()) : resolutionStrategyFactory;
-        DefaultConfiguration copiedConfiguration = instantiator.newInstance(DefaultConfiguration.class, domainObjectContext, newName, configurationsProvider, resolver, listenerManager,
-            metaDataProvider, childResolutionStrategy, fileCollectionFactory, buildOperationExecutor, instantiator, artifactNotationParser, capabilityNotationParser,
-            attributesFactory, rootComponentMetadataBuilder, documentationRegistry, userCodeApplicationContext, owner, projectStateRegistry, workerThreadRegistry,
-            domainObjectCollectionFactory, calculatedValueContainerFactory);
+        DefaultConfiguration copiedConfiguration = defaultConfigurationFactory.create(
```

Comment:
Is it safe to hold a reference to the factory like this and assume it can be used to create a copy with a new name?  Is the factory meant to be fully immutable?  Could any of the fields used as arguments when creating the new instance of `DefaultConfiguration` which now live in the factory class have changed between the construction of `this` and the call to `newConfiguration`, resulting a copy which isn't actually identical in the expected ways?